metabase/metabase
# initial commit

### Description

Adds the `TypeFilter` component as an isolated component with unit tests. This will be used in the `SearchFilterModal`, but for now I've added this as a separate PR for technical feedback before adding it to the integration branch.

This component uses the `useSearchListQuery` hook to fetch `available_models` using a general search query to get all of the possible entity types in their system. Then, it displays a grid of checkboxes of the translated name types for each element of `available_models`. Clicking a checkbox will call the `onChange` method which should return a list of `SearchModelType` that we will then be able to populate the URL with in the SearchApp in the future.

### How to verify

Run the unit tests to verify that everything passes. I've written more e2e tests that will be reviewed later. 

### Checklist
